From a841eeea1f3550d61a35ee87daca520bf951d1e2 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Wed, 3 Jan 2024 15:19:50 +0100
Subject: [PATCH 0854/1141] drm/panel: hx8394: don't reset the panel during the
 probe

To avoid to get a black screen after probe, don't reset the panel
& keep alive the panel (power supply & clocks) at least 60s.

Change-Id: I23d7a845f7523846a658e95b552dd909acd8db8b
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/352874
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/panel/panel-rocktech-hx8394.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/panel/panel-rocktech-hx8394.c b/drivers/gpu/drm/panel/panel-rocktech-hx8394.c
index 44111830d9cc..f2b87bf3df54 100644
--- a/drivers/gpu/drm/panel/panel-rocktech-hx8394.c
+++ b/drivers/gpu/drm/panel/panel-rocktech-hx8394.c
@@ -257,6 +257,7 @@ static int hx8394_unprepare(struct drm_panel *panel)
 	if (ret)
 		dev_warn(panel->dev, "failed to enter sleep mode: %d\n", ret);
 
+	pm_runtime_set_autosuspend_delay(ctx->dev, 1000);
 	pm_runtime_mark_last_busy(panel->dev);
 	ret = pm_runtime_put_autosuspend(panel->dev);
 	if (ret < 0)
@@ -337,7 +338,8 @@ static int hx8394_probe(struct mipi_dsi_device *dsi)
 			  MIPI_DSI_MODE_LPM | MIPI_DSI_CLOCK_NON_CONTINUOUS;
 
 	pm_runtime_enable(ctx->dev);
-	pm_runtime_set_autosuspend_delay(ctx->dev, 1000);
+	/* set delay to 60s to keep alive the panel to wait the splash screen */
+	pm_runtime_set_autosuspend_delay(ctx->dev, 60000);
 	pm_runtime_use_autosuspend(ctx->dev);
 
 	drm_panel_init(&ctx->panel, dev, &hx8394_drm_funcs,
@@ -397,8 +399,6 @@ static __maybe_unused int rocktech_hx8394_resume(struct device *dev)
 		return ret;
 	}
 
-	gpiod_set_value_cansleep(ctx->reset_gpio, 1);
-	mdelay(1);
 	gpiod_set_value_cansleep(ctx->reset_gpio, 0);
 	msleep(50);
 
-- 
2.39.2

